### PR TITLE
[FEAT] implement spacy models

### LINKED ISSUE #121: support spacy models
for NLP, it would be nice to add support for spacy models

### PR DESCRIPTION
**Pull request by isabelizimm**
closes #121

### CODE REVIEW

**Comment by isabelizimm** [comment]
at this state: figuring out how to handle non-serializable data types. this is implementing an interface to the `.predict()` methods on the Pipeline spaCy objects. Tests are not passing since spaCy requires a `spacy.tokens.Doc` object to make a prediction, which needs to be converted to json before POSTed to API endpoint. 

I could see that other frameworks have similar specific format. It might be useful to make the `predict` function single dispatch and then be able to register new data types in the spacy.py file.

File: vetiver/handlers/spacy.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            )
+        elif (
+            isinstance(self.prototype_data, pd.DataFrame)
+            and len(self.prototype_data.columns) != 1
+        ):  # is dataframe, more than one column
+            raise ValueError("Spacy prototype data must be a 1-column pandas DataFrame")
+        elif (
+            isinstance(self.prototype_data, dict) and len(self.prototype_data) != 1
+        ):  # is dict, more than one key
+            raise ValueError("Spacy prototype data must dictionary with 1 key")
```
Comment (isabelizimm) [inline vetiver/handlers/spacy.py:50]:
```suggestion
            raise ValueError("Spacy prototype data must be a dictionary with 1 key")
```

**Review by juliasilge — APPROVED** [review]
This worked well for me and looks good! 👍 I have two additional thoughts:

- (do now in this PR) can you update the README to include spaCy now?
- (let's think on this for future) how should we surface some model-specific documentation now that the number of supported models is growing?

**Comment by isabelizimm** [comment]
> This worked well for me and looks good! 👍 I have two additional thoughts:
> 
> * (do now in this PR) can you update the README to include spaCy now?

Added!

> * (let's think on this for future) how should we surface some model-specific documentation now that the number of supported models is growing?

That sounds like a great idea. Off the bat, the model handlers are shown in the docs. It might be advantageous to add examples of each model in the `examples/` directory, in docstrings, or both.

**Pull request merged**